change-address/REPWRITERSPECS/BANNO.CHANGE.ADDR.V1.POW

```diff
@@ -346,6 +365,331 @@ PRINT TITLE="BANNO.CHANGE.ADDR.V1.POW"
+    DO
+     SET EXTRAADDRESS TO UPPERCASE(NEWEXTRAADDR)
+     SET STREET TO UPPERCASE(NEWSTREET)
+     SET CITY TO UPPERCASE(NEWCITY)
+     SET STATE TO UPPERCASE(NEWSTATE)
+     SET ZIPCODE TO UPPERCASE(NEWZIPCODE)
+    END
+  END
+
+ IF FMERROR<>"" THEN
+  DO
+   RESPONSECODE=503
+   ERROR503CHR="Banno Chg Addr Err: FM-"+FMERROR
+   PARSEDTEXT(1)="Banno Chg Addr Err: FM-"+FMERROR
+   CALL CREATENOTERECORD
+  END
+ ELSE
+  DO
+   ADDRCHGSUCCESSFUL=TRUE
+   CALL POSTFMPROCESS
+  END [FMERROR=""]
+END [PROCEDURE]
+
+PROCEDURE GETCURRENTADDRESS
+[* Locate, by locator code, the current address to be updated. If
+** target locator code is 0, then target is the primary name record. Store
+** the record level where the targeted name record was found. As needed,
+** also store the parent record's ID or locator code to facilitate
+** updating the record.
+**
+** PASS    TARGETEDNAMELOC     The locator code of the name record to find
+** RETURN  TARGETNAMEFOUND     Was the target found? (boolean)
+** RETURN  TARGETNAMELEVEL     The record level where the name resides (i.e. SHARE, LOAN, EFT)
+** RETURN  TARGETNAMELEVELID   The ID of the name record's parent (i.e share ID)
+** RETURN  TARGETNAMELEVELLOC  The locator of the name record's parent (i.e. EFT locator)
+*]
+ OLDSTREET=""
+ OLDEXTRA=""
+ OLDCITY=""
+ OLDSTATE=""
+ OLDZIP=""
+ TARGETNAMEFOUND=FALSE
+ TARGETNAMELEVEL=""
+ TARGETNAMELEVELID=""
+ TARGETNAMELEVELLOC=0
+
+ IF TARGETEDNAMELOC=0 THEN
+  DO
+   FOR EACH NAME WITH (NAME:TYPE=0)
+    DO
+     TARGETNAMEFOUND=TRUE
+     TARGETNAMELEVEL="ACCOUNT"
+
+     OLDSTREET=NAME:STREET
+     OLDEXTRA=NAME:EXTRAADDRESS
+     OLDCITY=NAME:CITY
+     OLDSTATE=NAME:STATE
+     OLDZIP=SEGMENT(NAME:ZIPCODE,1,5)
+    END
+  END
+ ELSE [TARGETEDNAMELOC>0, Find the specific name locator to change]
+  DO
+   IF TARGETNAMEFOUND=FALSE THEN
+    DO
+[* Look in the account level
+*]
+     FOR EACH NAME WITH (NAME:LOCATOR=TARGETEDNAMELOC)
+      DO
+       TARGETNAMEFOUND=TRUE
+       TARGETNAMELEVEL="ACCOUNT"
+
+       OLDSTREET=NAME:STREET
+       OLDEXTRA=NAME:EXTRAADDRESS
+       OLDCITY=NAME:CITY
+       OLDSTATE=NAME:STATE
+       OLDZIP=SEGMENT(NAME:ZIPCODE,1,5)
+      END UNTIL TARGETNAMEFOUND=TRUE
+    END
+   IF TARGETNAMEFOUND=FALSE THEN
+    DO
+[* Look in the share level
+*]
+     FOR EACH SHARE
+      DO
+       TARGETNAMELEVELID=SHARE:ID
+       FOR EACH SHARE NAME WITH(SHARE NAME:LOCATOR=TARGETEDNAMELOC)
+        DO
+         TARGETNAMEFOUND=TRUE
+         TARGETNAMELEVEL="SHARE"
+
+         OLDSTREET=SHARE NAME:STREET
+         OLDEXTRA=SHARE NAME:EXTRAADDRESS
+         OLDCITY=SHARE NAME:CITY
+         OLDSTATE=SHARE NAME:STATE
+         OLDZIP=SEGMENT(SHARE NAME:ZIPCODE,1,5)
+        END UNTIL TARGETNAMEFOUND=TRUE
+      END UNTIL TARGETNAMEFOUND=TRUE
+    END
+   IF TARGETNAMEFOUND=FALSE THEN
+    DO
+[* Look in the IRS level
+*]
+     FOR EACH IRS
+      DO
```

Comment:
Hello @tkainz ,
 I think at line 567 you need "TARGETNAMELEVELLOC=IRS:LOCATOR"
Otherwise the code looks very good.

Reply:
It looks great, @tkainz ! Thanks for making the change